### PR TITLE
Add a generic Code of Conduct

### PR DESCRIPTION
**Pull request by TheBlueMatt**
No need to be too prescriptive, but examples are good. This is a pretty generic Code of Conduct that just says if you disrupt people's ability to contribute or harass someone, you will be shown the door.

It describes a Code of Conduct committee, which we'll need to define.

### CODE REVIEW

File: CoC.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * Unwelcomed sexual attention
+ * Inappropriate visual displays such as sexually-oriented or offensive photography, cartoons, drawings, or gestures
+ * Retaliation for reporting or threatening to report harassment
+
+Additionally, spam and other content which disrupts or prevents contributors from working is not acceptable.
+
+## The Code of Conduct Team
+
+A small team of contributors has volunteered to enforce this Code of Conduct. If you feel like a community member has engaged in inappropriate behavior, please don't hesitate to contact one of the following contributors via email:
+ * TODO
```
Comment (vincenzopalazzo) [inline CoC.md:26]: if you want you can add mine here

**Review by vincenzopalazzo — APPROVED** [review]
LGMT,

ACK https://github.com/lightning/bolts/pull/1207/commits/f23f55c12b100d6b0de7d6abf7d73fb6d2c8a140

**Review by Roasbeef — APPROVED** [review]

**Review by ProofOfKeags — APPROVED** [review]

**Comment by ariard** [comment]
I think this code of conduct is particularly weak, and we should rather take example on the moderation rules have they established on the bitcoin core side, cf. https://github.com/bitcoin-core/meta/blob/main/MODERATION-GUIDELINES.md

"A small team of contributors has volunteered to enforce this Code of Conduct. If you feel like a community member has engaged in inappropriate behavior, please don't hesitate to contact one of the following contributors via email:”

Is there an open nomination process ? Like done for BIP editors or maintainers of any true open-source software. Otherwise it’s not open-source but a closed door process where we make lenient and lazy agencements among friends.

"## The Code of Conduct Team’s Responsibilities”

Do the moderation team members have conflict of interests protections in the conduct of their professional activities ? E.g there was such section in the 2.2 of the bitmex open-source dev agreement (https://github.com/jonathanbier/Bitcoin-Developer-Grant-Agreement/blob/master/Bitcoin-Developer-Grant-Agreement.md) or few founders at blockstream inc had such “cannot be evil” protection. This a concern I’ve already echoed to TheBlueMatt, when Spiral popped up a code of conduct in the LDK a while back (https://github.com/lightningdevkit/lightningdevkit.org/pull/184).

Being constructive, I can communicate in private the conflict of interest protections that I consider acceptable, and that I’m usually including in my professional activities. They have been reviewed by my lawyers, and it should be easy to adapt them whatever you’re doing in life and receiving a monetary reward for activities related to bitcoin, be it employee, FOSS veteran, silicon valley startup founder, board member of a NYSE-listed company or a non-profit organization. They should be robust enough if you’re facing issues with a corporate entity or even governmental agency (e.g the FBI or whatever the law enforcement authority of the day). 

If this code of conduct is keeping to be pushed by a Block Inc employe in a more or less opaque fashion, I’ll see to engage in the future the personal responsibility of their CEO @jackjack in front of a US or EU civil or criminal court. Already removed the TheBlueMatt from future security vulnerabilities handling concerning the Lightning protocol, as I’m concerned with his personal ethics (and he’s doing the ostrich when you’re asking him clarification in private).

**Comment by ariard** [comment]
"_Being constructive, I can communicate in private the conflict of interest protections that I consider acceptable,_” (self-quote)

Reached out to Jess J from the Bitcoin Defense Legal Fund, as it’s the most appropriate channel to have an exchange of the interest protections. As they only work during open business days, this might take few weeks to proceed with such process.

I’ll let polite about what I think about the shortcoming of this current proposal. I can still see with one of my lawyer making an analysis of this proposal to see if it’s adapted to an international context of high-stake open-source softwares among stakeholders and developers of many cultural perspectives, and then publish this legal analysis in the public domain.

In a more constructive fashion, opened a counter-proposal based on the moderation rules as adopted in bitcoin core with #1208. Those rules are far more bright, neutral and consistent and to be clear I’m not the primary author of them. Looking forward for reviews and comments, at least if there are elements from this proposal that are justified to be added in the moderation rules proposal.

This is more constructive, rather than me going in the mainstream medias to point out in an extensive and exhaustive fashion all TheBlueMatt conflicts of interest since the start of his career in this space in 2011. The time of everyone is better allocated to go to fix common issues affecting all lightning softwares, especially mine.

This might take more time to settle on moderation rules in the lightning space, though in the meantime I think it’s okay to do it the way it was done in LDK pre-code of conduct. Stay patient, take on yourself and go to talk with whoever can have a repeated “aggressive" communication style towards other contributors addressing all the points, cf old 2020 [link](https://github.com/lightningdevkit/rust-lightning/pull/691#issuecomment-720539408).

File: CoC.md (hunk 1 above)
Comment (rustyrussell) [reply CoC.md:26]: Add me here, that's fine.

**Review by rustyrussell — APPROVED** [review]
"Be excellent to each other"

**Comment by ariard** [comment]
@rustyrussell Are you receiving a financial compensation from or promise of such by Block Inc to ACK this PR ? Or did you have discussions with them to consider them as an employment fallback plan, like Greg Sanders by transitioning from Blockstream Lightning team to Spiral Wizards in 2023 ? Or at least do you have minimal conflict of interests protection in your work contract with Blockstream Inc, as some original founders users to have ?

Be sure by asking those questions, I’m just following your words “_let’s make sure we keep each other honest, ok ?_” from your personal essay on the “The Corrosion of Ethics in Cryptocurrencies” (available here: https://rusty.ozlabs.org/2018/02/14/the-corrosion-of-ethics-in-cryptocurrencies.html)

**Comment by ariard** [comment]
Making my critics and concerns more verbose about this current proposal, as I think it's going to far in the direction "to preach or try to control behavior" which is explicitly ruled out by the actual bitcoin core moderation rules. The enumeration of harassment bullet points does not give any constitutive elements, there is no oversight of the code of conduct team and
the list of questions is completely subjective (thanks no thanks I don't wish to be responsible of TheBlueMatt's mental health because he can be upset I’m telling him too often his rust-lightning code is broken). This code of conduct doesn't explain how we're supposed a code of conduct team member himself misbehaving (what if e.g TheBlueMatt had some romantic relationship with someone in the lightning space back in 2018 and this interferes with his judgement during the conduct of a code of conduct janitorial task ?).

Further, this is open a wide door for the code of conduct team to preach in people's private life, without their explicit authorization. Back in late 2020, I had a veteran bitcoin core contributor litterally pressuring on some aspects of my private life to get my support and involvements in the launch of a non-profit bitcoin FOSS organization at a time he was still fundraising for it (with a clearly non-sollicited call in date of Sunday 22 November 2020 in that sense). After multiple refusals from his side to clarify his management philosophy or take safe-guard dispositions to prevent future ethical mishandling,
I have to cut with this professional project, as I was deeply concerned he would keep pressuring some aspects of my private life to coerce my open-source works and writing in matters of bitcoin technical consensus, or even worst bitcoin or lightning 
security vulnerabilities handling.

This is very unclear to me why Matt Corallo is proposing this code of conduct (is it by ideology blindness ?), if he's under strain from his current employer, which might also financially inciting the other regular lightning protocol contributors to ACK it (for an external observer - there are no written motivations why they ACKing  it). Due to the previous incident of the misleading establishment of the code of conduct in LDK in 2022 to silence technically critical voices about LDK codebase, the new incident with the lightning summit in Japan where some people are cancelled from the invitation list, and this new ill-intended code of conduct, if this is one he's adopted I'll consider legal actions against Matthew Corallo, and his two directly hierarchical responsible Mr. Lee and Mr. Dorsey as I'm deeply concerned it could be instrumentalized in the future to harm
lightning contributors private lifes (nota bene: no concern to sue a tech titan, already meet others far more impressive in the past). If some people estimate themselves entitled to try to control or judge people behaviors in a public domain, be ready to face real professional judges (-- crossing the line of issuing legal threat in public toward other open-source was crossed first by TheBlueMatt when establishing the LDK [code of conduct](https://github.com/lightningdevkit/lightningdevkit.org/pull/184#issuecomment-1368126430))

Zooming out, this new code of conduct is interferring with one ability to contribute on the Lightning protocol, especially in matters of proposing spec improvement to lightly fix some vulnerabilities affecting the whole network of lightning nodes. Or more deeply affecting deep lightning security issues interdepend with the consensus rules affecting the > $1 TB underlying bitcoin base layer. At the end of the day, one has to put the interest of the end-users (“the anonymous pleb") above the reputation or the career of few corrupted open-source maintainers. Lightning and bitcoin are used daily by many people over the world as reliable tools to underpin their financial lifes, be it in developing countries, war zones or even in inflation-ravaged more traditional countries, their interests matters more than the one of few "insiders", and their deserve a scalable and secure Lightning protocol.

**Comment by ariard** [comment]
_For authenticity purpose - This thread is timestamped in the bitcoin blockchain itself secured by a lot of EH / second, which is itself a harder to forge as a technical medium than a cloud-powered Github repository, that can be tampered by untrustworthy admins._

**Pull request merged**